Buildings/Controls/OBC/CDL/Logical/Validation/Timer.mo

```diff
@@ -1,21 +1,52 @@
 within Buildings.Controls.OBC.CDL.Logical.Validation;
 model Timer "Validation model for the Timer block"
 
-  Buildings.Controls.OBC.CDL.Logical.Sources.Pulse booPul(
-      width=0.5, period=2) "Block that outputs cyclic on and off"
-    annotation (Placement(transformation(extent={{-40,-10},{-20,10}})));
+
   Buildings.Controls.OBC.CDL.Logical.Timer resetTimer(reset=true)
     "Timer will reset"
-    annotation (Placement(transformation(extent={{20,-10},{40,10}})));
+    annotation (Placement(transformation(extent={{20,60},{40,80}})));
   Buildings.Controls.OBC.CDL.Logical.Timer accuTimer(reset=false)
-    "Timer will not reset"
-    annotation (Placement(transformation(extent={{20,-50},{40,-30}})));
+    "Reset timer based on boolean input"
+    annotation (Placement(transformation(extent={{20,20},{40,40}})));
+  Buildings.Controls.OBC.CDL.Logical.Timer accuTimer1(reset=false)
+    "Reset timer based on boolean input"
+    annotation (Placement(transformation(extent={{20,-20},{40,0}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Sources.Pulse booPul1(
+    final width=0.1,
+    final period=4)
+    "Block that outputs cyclic on and off"
+    annotation (Placement(transformation(extent={{-40,20},{-20,40}})));
+  Buildings.Controls.OBC.CDL.Continuous.GreaterEqualThreshold greEquThr(
+    final threshold=1.5)
+    annotation (Placement(transformation(extent={{-80,-60},{-60,-40}})));
+  Buildings.Controls.OBC.CDL.Logical.Pre pre
+    annotation (Placement(transformation(extent={{-40,-60},{-20,-40}})));
+  Buildings.Controls.OBC.CDL.Logical.Sources.Pulse booPul(
+    final width=0.5,
+    final period=2) "Block that outputs cyclic on and off"
+    annotation (Placement(transformation(extent={{-40,60},{-20,80}})));
```

Comment:
There must also be a test with `reset=true` and `u0` becoming `true` while `u` is `true`.
The expected behavior is that `y` is reset to `0` and then keeps increasing.